VACMS-8149: Add 2:3 crop style and update 2:3 thumbnail style.

msbtterswrth opened the pull request:
## Description

Closes #8149.

## Testing done


## Screenshots
![image](https://user-images.githubusercontent.com/8375335/155614641-d8cf4bfa-138f-43dd-9fdc-a7c749728709.png)
![image](https://user-images.githubusercontent.com/8375335/155614843-93c17dbc-4d94-4afb-ab97-f822f8a13918.png)


## QA steps

As a cms editor
1. I can see and select the crop type 2:3 when inserting media
2. I can confirm on the frontend that the 2:3 crop is applied to media


### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`


### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

kevwalsh commented on config/sync/image.style.2_3_medium_thumbnail.yml:
```diff
@@ -3,7 +3,7 @@ langcode: en
 dependencies:
   config:
-    - crop.type.3_2
+    - crop.type.1_1_5
```
i think it would be a lot easier to keep machine names standardized around integers where possible, so 2_3

msbtterswrth replied:
fwiw, i didn't create those, that's what was already there for that crop type. i 100000% agree with you though.

kevwalsh replied:
oh yeah i see that now. OK i'm fine with just reusing what's there.

msbtterswrth replied:
i ended up creating a new crop style, and renaming the old one to 1:1.5. We can follow up later with a PR to remove it

kevwalsh commented on config/sync/image.style.crop_2_3.yml:
```diff
@@ -0,0 +1,18 @@
+status: true
+dependencies:
+  config:
+    - crop.type.1_1_5
```
should be crop.type.2_3

msbtterswrth replied:
updated

kevwalsh commented on config/sync/image.style.crop_2_3.yml:
```diff
@@ -0,0 +1,18 @@
+    id: crop_crop
+    weight: 1
+    data:
+      crop_type: '1_1_5'
```
should be `'2_3'`

msbtterswrth replied:
updated

kevwalsh commented on config/sync/image.style.2_3_medium_thumbnail.yml:
```diff
@@ -14,7 +14,8 @@ effects:
     weight: 1
     data:
-      crop_type: '3_2'
+      crop_type: '1_1_5'
```
should be 2_3

msbtterswrth replied:
updated

kevwalsh reviewed (changes requested):
LOL at the example. 

i'd like to stick to integers for consistency (mirrors the 3:2 landscape crop and image style)

kevwalsh reviewed (changes requested):
@msbtterswrth on second thought, it looks like a one line change to update 1_1_5 to 2_3 at /config/sync/crop.type.1_1_5.yml

swirtSJW reviewed (commented):
Config changes look good.

Is this related or closes?

This needs some more specific on QA steps.
![image](https://user-images.githubusercontent.com/5752113/155660676-73283156-e741-4cb9-a49c-44fed85414ff.png)

   We want to test functionality, not just go observe the setting exists.   How should a QA person evaluate this to see if it actually works?

swirtSJW commented:
Behat failures are related:

```
 +++ Unexpected image style (present in Drupal, absent from specification)
      | Cropped: 2:3 | crop_2_3 | 
```
and
```
 --- Missing image effect (present in specification, absent from Drupal)
      | 2:3 medium thumbnail | Manual crop | uses 3:2 crop type |
      +++ Unexpected image effect (present in Drupal, absent from specification)
      | 2:3 medium thumbnail | Manual crop | uses 2:3 crop type |
      | Cropped: 2:3         | Manual crop | uses 2:3 crop type | 
```

You can test these locally with
`ddev behat --tags=image_styles`
`ddev behat --tags=image_effects`

msbtterswrth commented:
Thanks for those test scripts! tests are updated and running now, @swirtSJW. 

I also fixed the closes, and added an additional qa step, though i think the frontend was working anyways. I think the admin is showing wrong crop styles in several instances which has a separate ticket that we can address that in I believe.

swirtSJW reviewed (approved):
Nice work.
